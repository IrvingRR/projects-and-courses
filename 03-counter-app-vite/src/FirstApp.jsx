import PropTypes from 'prop-types';
import { Fragment } from 'react';

const FirstApp = ({ title, subtitle }) => {
  return (
    <Fragment>
        <h1 data-testid='test-title'>{ title }</h1>
        <p>{ subtitle }</p>
    </Fragment>
  );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
};

FirstApp.defaultProps = {
    subtitle: 'No subtitle...'
}

export default FirstApp;