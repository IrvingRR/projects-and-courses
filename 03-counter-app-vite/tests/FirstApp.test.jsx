import { render } from '@testing-library/react';
import FirstApp from '../src/FirstApp';

describe('Tests in <FirstApp/>', () => {

    test('Should to do match with the snapshot', () => {
        const title = 'Hi!, frontend developer';
        const { container } = render(<FirstApp title={ title }/>);

        expect(container).toMatchSnapshot();

    });

    test('should to show the title in a h1 tag', () => { 
        const title = 'Hi developer!';
        const { getByText, getByTestId } = render(<FirstApp title={ title }/>);
        expect(getByText(title)).toBeTruthy();
        expect(getByTestId('test-title').innerHTML).toContain(title);
    });

    

});
