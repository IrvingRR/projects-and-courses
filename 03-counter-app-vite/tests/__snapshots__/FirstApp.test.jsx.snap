// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp/> Should to do match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hi!, frontend developer
  </h1>
  <p>
    No subtitle...
  </p>
</div>
`;
